components/raftstore/src/store/peer.rs

```diff
@@ -1830,8 +1856,8 @@ where
     fn apply_reads<T>(&mut self, ctx: &mut PollContext<EK, ER, T>, ready: &Ready) {
         let mut propose_time = None;
         let states = ready.read_states().iter().map(|state| {
-            let uuid = Uuid::from_slice(state.request_ctx.as_slice()).unwrap();
-            (uuid, state.index)
+            let read_index_ctx = ReadIndexContext::parse(state.request_ctx.as_slice()).unwrap();
+            (read_index_ctx.id, read_index_ctx.locked, state.index)
```

Comment:
`ReadIndexContext` may break compatibility. If an old version tikv received `ReadIndexContext` and try to parse it with [`Uuid::from_slice`](https://docs.rs/uuid/0.8.1/uuid/struct.Uuid.html#method.from_slice) then the `unwrap()` here will panic

Reply:
There won't be such a case.

1. If read index is sent by a new version TiKV and with the new format
  1.1 If the leader is new version, the response will have the correct info.
  1.2 If the leader is old version, the context will be the same as before. We can also parse it, but `locked` will not be set. But it does not matter because TiDB upgrades later than TiKV so locks needn't be checked.
2. If read inex is sent by an old version TiKV with old format, `request` will be `None` and the leader won't set other fields for this context. The old version TiKV will still get a UUID context.

Reply:
> If read inex is sent by an old version TiKV with old format, request will be None and the leader won't set other fields for this context. The old version TiKV will still get a UUID context.

I see, but it will be good to have a unit test about `ReplicaReadLockChecker:: on_step` to ensure it though.

Reply:
Makes sense. Let me add it.